pkg/client/logs.go

```diff
@@ -111,30 +140,65 @@ func (r *Reader) Read(p []byte) (int, error) {
 	return len(data), nil
```

Comment:
Should we be returning `r.err` here? In the case where there was an error, we treat the error message as `data` which is what we copy to `p` and return the length of here, so we would be providing the contents of the error, but not the actual error itself.

Reply:
No, in case there is overflow data we want to not return the error until all the data is done being sent.

So what we're doing now is printing _all_ the data from the bystream now and mixing in the printed errors as they arrive. However we don't return an `error` object until all the data is printed so that someone can still do things like reading until encountering an error.

We don't want to return an error until after all data is printed or else 1 broken set of logs will prevent us from getting all the other logs.

Reply:
Yep, but there are separate paths for when the data doesn't fit into the buffer (just above this return and also at the beginning of the function when we prioritise the overflow buffer) so that's already being taken into consideration. The reason this was confusing me is that in [`TestLateErrors`](https://github.com/vmware-tanzu/sonobuoy/pull/1007/files#diff-b3336300806d23af14262c2b62843558R63-R65) we are now making 5 calls to `Read`, so we print out all of the error, but don't get the actual error until the final call when there is no data from either stream. We want to entirely empty the bytestream before any returning any error. I've messed around with a test and I think this is making sense now :) Sorry I'm being so slow with this!